torch/distributed/fsdp/_common_utils.py

```diff
@@ -68,7 +68,10 @@ def _is_composable(state: _FSDPState):
 
 
 @no_type_check
-def _all_handles(state: _FSDPState) -> List:
+def _all_handles(state: _FSDPState) -> List[flat_param_file.FlatParamHandle]:
```

Comment:
I think this function needs to be changed for manual "wrapping" support since we cannot directly return `state._handles` but instead need to collect all of the handles, accounting for nested applications of `fully_shard` (analogously to the wrapper path).